pkg/handler/common/provider/aws.go

```diff
@@ -222,36 +227,64 @@ func SetDefaultSubnet(machineDeployments *clusterv1alpha1.MachineDeploymentList,
 	return subnets, nil
 }
 
-func AWSSizes(region string) (apiv1.AWSSizeList, error) {
+func AWSSizes(region string, quota kubermaticv1.MachineDeploymentVMResourceQuota) (apiv1.AWSSizeList, error) {
 	if data == nil {
 		return nil, fmt.Errorf("AWS instance type data not initialized")
 	}
 
 	sizes := apiv1.AWSSizeList{}
 	for _, i := range *data {
-		// TODO: Make the check below more generic, working for all the providers. It is needed as the pods
-		//  with memory under 2 GB will be full with required pods like kube-proxy, CNI etc.
-		if i.Memory >= 2 {
-			pricing, ok := i.Pricing[region]
-			if !ok {
-				continue
-			}
+		pricing, ok := i.Pricing[region]
+		if !ok {
+			continue
+		}
+		price := pricing.Linux.OnDemand
+		// not available
+		if price == 0 {
+			continue
+		}
 
-			// Filter out unavailable or too expensive instance types (>1$ per hour).
-			price := pricing.Linux.OnDemand
-			if price == 0 || price > 1 {
-				continue
-			}
+		sizes = append(sizes, apiv1.AWSSize{
+			Name:       i.InstanceType,
+			PrettyName: i.PrettyName,
+			Memory:     i.Memory,
+			VCPUs:      i.VCPU,
+			GPUs:       i.GPU,
+			Price:      price,
+		})
+	}
+
+	return filterAWSByQuota(sizes, quota), nil
+}
+
+func filterAWSByQuota(instances apiv1.AWSSizeList, quota kubermaticv1.MachineDeploymentVMResourceQuota) apiv1.AWSSizeList {
+	filteredRecords := apiv1.AWSSizeList{}
+
+	// Range over the records and apply all the filters to each record.
+	// If the record passes all the filters, add it to the final slice.
+	for _, r := range instances {
+		keep := true
+
+		// Filter too expensive instance types (>1$ per hour) if GPU not enabled
+		if !quota.EnableGPU && r.Price > 0 {
```

Comment:
`r.Price > 1` should be here.

BTW should we still filter these (>1$ per hour), I am not sure if users are aware of this restriction. Or maybe we could add another filter for price (in a different ticket).

Reply:
There was a request for this some time ago, so I decided to leave this.